src/_pytest/capture.py

```diff
@@ -692,6 +704,39 @@ def snap(self):
         return res
 
 
+class OrderedCapture(SysCapture):
+    """This class uses deque to keep streams order."""
+    streams = collections.deque()
+
+    def _get_writer_class(self):
+        return OrderedWriter(self.fd)
+
+    def snap(self):
+        res = self.tmpfile.getvalue()
+        if self.name == "stderr":
+            # both streams are being read one after another, while stderr is last - it will clear the queue
+            self.streams.clear()
+        return res
+
+
+class OrderedWriter:
+    encoding = sys.getdefaultencoding()
+
+    def __init__(self, fd):
+        super().__init__()
+        self._fd = fd  # type: int
+
+    def write(self, text, **kwargs):
+        OrderedCapture.streams.append((text, self._fd))
```

Comment:
Shouldn't use a class attribute, should it?

Reply:
I did that on purpose, but I'm not sure how it will work with stacking. E.g. when a broader-scoped fixture will use `capsys`, how it will interfere with the one in a test. I'll test it out.

Reply:
I did some tests, and it seems to work exactly like the current `capsys`. Having the `deque` as a class attribute allows us to just reference it without the need to passing it as an argument.